platform/ios/src/MGLMapView.h

```diff
@@ -526,6 +526,17 @@ MGL_EXPORT IB_DESIGNABLE
  */
 @property(nonatomic, getter=isHapticFeedbackEnabled) BOOL hapticFeedbackEnabled;
 
+/**
+ A Boolean value that determines whether the updating pitch will also affect the altitude.
+ 
+ Setting this property to `NO` will allow animation libraries outside of this SDK to control the
+ map camera without the built-in side effect that pitch influences altitude. This will make the
+ results of using outside animation more predictable and easier to control.
+ 
+ The default value of this property is `YES`.
+ */
+@property(nonatomic, getter=isCameraAltitudeAffectedByPitch) BOOL cameraAltitudeAffectedByPitch;
```

Comment:
MGLMapView’s public API is expressed in terms of zoom levels, not in terms of altitude. Generally speaking, MGLMapView’s direct parameters are tied to 2D concepts for simplicity, while MGLMapCamera is better suited for expressing 3D concepts because it can account for two different frames of reference (viewpoint versus focal point). Mixing the two concepts would promote one frame of reference over the other, which would call into question the very purpose of having MGLMapCamera as a separate class.

As far as MGLMapView is concerned, pitch is a detail of MGLMapCamera. You’ll note that the rest of MGLMapView avoids referring to pitch, because normally the pitch is intertwined with other camera parameters and the camera’s frame of reference.

Reply:
Right now if I want to change pitch I have to use MGLMapCamera. When the MGLMapCamera is passed to cameraOptionsObjectForAnimatingToCamera, mbgl updates it's state with the zoomLevel calculated from MGLZoomLevelForAltitude.

All of the setCamera functions use cameraOptionsObjectForAnimatingToCamera.

Reply:
My point is that MGLMapView’s API shouldn’t even mention pitch. If mentions pitch, it should provide the full complement of ways to interact with the pitch, which it doesn’t by design.

Think of camera as a set of parameters that get updated together. They have to be updated together in order to maintain consistency. If you want to update the pitch without updating other parameters, you set the other parameters to special values. For example, set heading to -1 or centerCoordinate to an invalid coordinate.